app/jobs/globus_list_job.rb

```diff
@@ -28,27 +28,24 @@ def perform(content:, cancel_check_interval: 100)
 
   attr_reader :content
 
-  def list_files
-    endpoint_client.list_files.each_with_index do |file_info, index|
+  def list_files # rubocop:disable Metrics/AbcSize
+    GlobusClient.list_files(user_id: user.email_address, path: globus_path, notify_email: false)
+                .each_with_index do |file_info, index|
       # Only check for cancel every 100 files
       break if (index % @cancel_check_interval).zero? && canceled?
 
-      filepath = file_info.name.delete_prefix(globus_path)
+      filepath = Pathname.new(file_info.name).relative_path_from(globus_absolute_path).to_s
```

Comment:
Defer path manipulation to the `Pathname` library instead of manually string munging.